developers/weaviate/more-resources/dspy.md

```diff
@@ -0,0 +1,62 @@
+---
+title: DSPy Integration
+sidebar_position: 5
+image: og/docs/more-resources.jpg
+---
+[DSPy](https://github.com/stanfordnlp/dspy) is a framework for programming language models created by Stanford NLP. DSPy introduces two key concepts: **programming model** and **optimizers**.
+
+- **Programing model**: The programming model lets you define a series of components that make a language model request. Components include input and output fields, task descriptions, and calls to a vector database like Weaviate.
+
+- **Optimizers**: Optimizers compile your DSPy program to tune the language model prompt and/or the weights.
+
+## DSPy and Weaviate
+
+Weaviate is integrated with DSPy through the retriever module! You only need to connect to your Weaviate cluster through [WCS](https://console.weaviate.cloud/) or localhost, and pass in your collection:
+
+```python
+weaviate_client = weaviate.Client("http://localhost:8080") # or pass in your WCS cluster url
+
+retriever_module = WeaviateRM("WeaviateBlogChunk", # collection name
+                    weaviate_client=weaviate_client)
+```
+
+## Our Resources 
+Here are a few resources on using DSPy from the Weaviate team!
+
+The resources are broken into two categories: 
+1. [**Hands on Learning**](#hands-on-learning): Content framed to build your technical understanding with end-to-end tutorials. 
+
+2. [**Read and Listen**](#read-and-listen): Content designed to help develop your conceptual understanding of these technologies.
+
+
+### Hands on Learning 
+
+| Topic | Details |
+| --- | --- |
+| **Getting Started with RAG in DSPy** | Learn about the four components needed to build a DSPy program: Installation, settings, Datasets with `dspy.Example`, LLM Metrics, The DSPy programming model, and Optimization. [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/dspy/1.Getting-Started-with-RAG-in-DSPy.ipynb) and [Video](https://youtu.be/CEuUG4Umfxs?si=4Gp8gR9glmoMJNaU) |
+| **DSPy + Weaviate for the Next Generation of LLM Apps** | Build a 4-layer DSPy program for generating blog posts from queries. [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/dspy/2.Writing-Blog-Posts-with-DSPy.ipynb) and [Video](https://youtu.be/ickqCzFxWj0?si=AxCbD9tq2cbAH6bB)|
+| **RAG with Persona** | Build a compound AI system with DSPy, Cohere, and Weaviate, adding persona to language model outputs. [Notebook](https://github.com/weaviate/recipes/tree/main/integrations/dspy/fullstack-recipes/RAGwithPersona) and [Post](https://twitter.com/ecardenas300/status/1765444492348243976)|
```

Comment:
Is persona a product? This should probably be: 

adding Persona  (product)
or
adding a persona  (not a product)

Reply:
Agreed! Switched to: 
"Build a compound AI system with DSPy, Cohere, and Weaviate, where you'll add a persona to the language model."